src/theia/sfm/reconstruction_builder.h

```diff
@@ -147,7 +147,8 @@ class ReconstructionBuilder {
                        const ImagePairMatch& matches);
 
   // Assign a mask for each image.
-  bool SetMasksForFeaturesExtraction(const std::vector<std::string>& mask_filepaths);
+  bool SetMasksForFeaturesExtraction(
```

Comment:
I would prefer if you use the same interface as AddMaskForFeaturesExtraction where the caller provides the mapping between image and image masks, adding one image to mask mapping at a time.  Otherwise, this interface assumes a very specific naming format for the image masks which can lead to trouble. It is better in my experience to only make that assumption at the "application" level (i.e. in the reconstruction builder).

In the build_reconstructions application we can take a container of image masks (found from the image mask wildcard), find the corresponding image, then add the masks to the reconstruction builder.